packages/react-components/react-tags/src/components/Tag/Tag.types.ts

```diff
@@ -1,20 +1,46 @@
+import { AvatarSize, AvatarShape } from '@fluentui/react-avatar';
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
-import { Avatar } from '@fluentui/react-avatar';
+import { Button } from '@fluentui/react-button';
+
+export type TagContextValues = {
+  avatar: {
+    size?: AvatarSize;
+    shape?: AvatarShape;
+  };
+};
 
 export type TagSlots = {
   root: NonNullable<Slot<'div'>>;
-  content?: Slot<'span'>;
-  avatar?: Slot<typeof Avatar>;
-  icon?: Slot<'span'>;
-  primaryText?: Slot<'span'>;
-  secondaryText?: Slot<'span'>;
-  dismissButton?: NonNullable<Slot<'button'>>;
+
+  /**
+   * Slot for an icon or other visual element
+   */
+  media: Slot<'span'>;
+
+  /**
+   * A layout wrapper for the icon slot, the primaryText and secondaryText slots
+   */
+  content: Slot<'div'>;
+
+  icon: Slot<'span'>;
```

Comment:
There's both a media slot whose description is 'slot for an icon or other visual element' and also an icon slot

is there any way we could make this naming a bit clearer?

Reply:
The only reason I have icon as a separate slot is that I can't figure out a style for one slot. Avatar has 8px gap to the text while icon have 2px:
<img width="284" alt="Screenshot 2023-05-01 at 20 33 32" src="https://user-images.githubusercontent.com/28751745/235452223-8d98118d-519f-44f3-b101-47324d6767de.png">

Reply:
Oh then I completely misunderstood the purpose of this slot: we should not have it . if someone puts an icon in the media slot... they have broken all design 🥳🥳🥳

Reply:
I brought this up with design, Toshie will try to figure out if there's a solution to make avatar and icon fit in one slot. In the meantime, I'll use bernardo's idea and fit them into one gridArea :)

Reply:
> I'll start by saying I'm not sure if this is a good idea :)
> 
> What do you think about defaulting to the icon spacing and then using the Avatar context to override the spacing for Avatars?

@spmonahan I don't like two slots either :D But I don't have a good way of doing style overrides. I chat with Designer on this issue to see if we can adjust the design to fit both avatar/icon in one slot.
I've tried having `className` on AvatarContext (#27755) and use it to apply style override for avatar on top of icon slot. But it was point out to me that AvatarContext is not supposed to be used this way. TBH it feels weird to me as well.